### ISSUE TITLE
Recommend eslint:no-label for functional setState

### ISSUE THREAD

**Issue by tsiq-swyx** [reactjs/reactjs.org#1004]
this is a pretty common mistake when using the functional form of setState:

![image](https://user-images.githubusercontent.com/35976578/41931269-96b574c2-794b-11e8-8766-878c0d73b951.png)

and we have no warning in React (yet) for this when it happens. easiest guard is having eslint:no-label tell you about it. Should we add a mention in [the setState docs](https://reactjs.org/docs/faq-state.html#what-is-the-difference-between-passing-an-object-or-a-function-in-setstate)?

**Comment by alexkrolick** [comment]
Maybe add a note here? https://reactjs.org/docs/faq-state.html#what-is-the-difference-between-passing-an-object-or-a-function-in-setstate

**Comment by tsiq-swyx** [comment]
yeah. i'm also trying to add a warning in react itself: https://github.com/facebook/react/issues/13111

**Comment by rickhanlonii** [comment]
On the new site I'm not sure it makes sense to recommend the lint rule, especially in 2025 where types and strict mode should catch this, but it would probably be good to have a Pitfall callout or Troubleshooting item here: https://react.dev/reference/react/useState#troubleshooting
